AutomatedTesting/Gem/PythonTests/scripting/NodeCategory_ExpandOnClick.py

```diff
@@ -45,79 +55,90 @@ def NodeCategory_ExpandOnClick():
+    async def run_test(self):
 
-    # 1) Open Script Canvas pane
-    general.open_pane("Script Canvas")
-    Report.critical_result(Tests.pane_open, general.is_pane_visible("Script Canvas"))
+        # Preconditions
+        general.idle_enable(True)
+        general.close_pane(SCRIPT_CANVAS_UI)
+        helper.wait_for_condition(lambda: general.is_pane_visible(SCRIPT_CANVAS_UI) is None, WAIT_TIME_3)
 
-    # 2) Get the SC window objects
-    editor_window = pyside_utils.get_editor_main_window()
-    sc = editor_window.findChild(QtWidgets.QDockWidget, "Script Canvas")
-    if sc.findChild(QtWidgets.QDockWidget, "NodePalette") is None:
-        action = pyside_utils.find_child_by_pattern(sc, {"text": "Node Palette", "type": QtWidgets.QAction})
-        action.trigger()
-    node_palette = sc.findChild(QtWidgets.QDockWidget, "NodePalette")
-    nodeTree = node_palette.findChild(QtWidgets.QTreeView, "treeView")
-    ai_index = pyside_utils.find_child_by_pattern(nodeTree, "AI")
+        # 1) Open Script Canvas pane
+        general.open_pane(SCRIPT_CANVAS_UI)
+        Report.critical_result(Tests.pane_open, general.is_pane_visible(SCRIPT_CANVAS_UI))
 
-    # 3) Ensure all categories are collapsed for a clean state
-    nodeTree.collapseAll()
+        # 2) Get the SC window objects
+        scripting_tools.initialize_editor_object(self)
+        scripting_tools.initialize_sc_editor_objects(self)
+        if self.sc_editor.findChild(QtWidgets.QDockWidget, NODE_PALETTE_QT) is None:
+            action = pyside_utils.find_child_by_pattern(self.sc_editor, {"text": NODE_PALETTE_UI, "type": QtWidgets.QAction})
+            action.trigger()
+        #wait for the node palette and other SC elements to render
+        helper.wait_for_condition(lambda: self.sc_editor.findChild(QtWidgets.QDockWidget, NODE_PALETTE_QT) is not None, WAIT_TIME_3)
+        node_palette_widget = self.sc_editor.findChild(QtWidgets.QDockWidget, NODE_PALETTE_QT)
+        node_palette_node_tree = node_palette_widget.findChild(QtWidgets.QTreeView, TREE_VIEW_QT)
+        node_palette_math_category = pyside_utils.find_child_by_pattern(node_palette_node_tree, NODE_CATEGORY_MATH)
 
-    # 4) Left-Click on a node category arrow to expand it
-    left_click_arrow(nodeTree, ai_index)
+        # 3) Ensure all categories are collapsed for a clean state
+        node_palette_node_tree.collapseAll()
 
-    # 5) Verify it expanded
-    Report.result(Tests.click_expand, nodeTree.isExpanded(ai_index))
+        # 4) Left-Click on a node category arrow to expand it
+        self.left_click_expander_button(node_palette_node_tree, node_palette_math_category)
 
-    # 6) Left-Click on a node category arrow to collapse it
-    left_click_arrow(nodeTree, ai_index)
+        # 5) Verify it expanded
+        category_has_expanded = helper.wait_for_condition(lambda: node_palette_node_tree.isExpanded(node_palette_math_category), WAIT_TIME_3)
+        Report.result(Tests.click_expand, category_has_expanded)
```

Comment:
Combine action and verification under one comment/step and updated test steps in desc.

Reply:
combined and updated comments